src/core/utils.dart

```diff
@@ -1 +1,141 @@
-// TODO: Add any utility functions here
\ No newline at end of file
+import 'note_list.dart';
+import 'dart:io';
+
+// returns a note index in the list
+// get from either a string search that searches note title and content, or from note ID in list
+int getNoteIndexFromInput(NoteList list, String prompt,
+    {bool displayNote = true}) {
+  while (true) {
+    // Establish input
+    String input = getStringInput(prompt +
+        " (Search by typing the note's content or title, alternatively type the note's ID) ");
+
+    try {
+      // Try searching through integer ID by parsing input
+      // Throws exception if input is not integer
+      int inputInt = int.parse(input);
+
+      // No exception thrown, which means input is integer
+      // Menu displays indices incremented by 1, so we compensate by subtracting 1 from input to get index
+      int index = inputInt - 1;
+
+      // If the specified index in the note list exists, we return that index
+      // If it doesn't exist, we just re-iterate the loop
+      if (checkNoteAtIndexExists(list, index)) {
+        if (displayNote) {
+          print("\n================");
+          print(list.notes[index].toString());
+          print("================\n");
+        }
+        return index;
+      }
+    } catch (e) {
+      // Exception thrown, search by string input instead
+
+      // function returns -1 if note does not exist
+      int index = getNoteIndexFromStringSearch(list, input);
+
+      // If index equals -1, it means it does not exist. In this case we re-iterate the loop
```

Comment:
It would have been more user-friendly if it exited the menu or provided an option to exit, as users can get stuck in a loop.
![Screen Shot 1445-07-21 at 3 56 50 PM](https://github.com/GDSC-IAU/Selection-Project/assets/86604450/3071a358-2dc5-4f20-b0d0-e4ba53182ccf)